backend/lib/endpoints/reports.js

```diff
@@ -40,15 +48,599 @@ async function routes(app) {
+            log.elapsedTimeText = {
+              created: translateISOtoRelativeTime(log.createdAt),
+            };
+          });
+          return logs;
+        }),
+      );
+
+      if (logsErr) {
+        req.log.error(logsErr, "Failed requesting logs");
+        throw app.httpErrors.internalServerError();
+      } else if (logs === null) {
+        return {
+          logs: [],
+          meta: {
+            total: 0,
+          },
+        };
+      } else {
+        return {
+          logs,
+          meta: {
+            total: totalLogsCount || 0,
+          },
+        };
+      }
+    },
+  );
+
+  app.get(
+    "/stats",
+    {
+      preValidation: [
+        app.authenticate,
+        app.setActor,
+        app.checkScopes([SCOPES.STATS_READ]),
+      ],
+    },
+    async (req) => {
+      const aggregationPipeline = [
+        {
+          $facet: {
+            pendingPostsCount: [
+              {
+                $match: { status: "flagged" },
+              },
+              {
+                $group: {
+                  _id: null,
+                  count: { $sum: 1 },
+                },
+              },
+            ],
+            removedPostsCount: [
+              {
+                $match: { status: "removed" },
+              },
+              {
+                $group: {
+                  _id: null,
+                  count: { $sum: 1 },
+                },
+              },
+            ],
+            keptPostsCount: [
+              {
+                $match: {
+                  status: "public",
+                  reportedBy: { $exists: true, $not: { $size: 0 } },
+                },
+              },
+              {
+                $group: {
+                  _id: null,
+                  count: { $sum: 1 },
+                },
+              },
+            ],
+            dailyNewReportsCounts: [
+              {
+                $match: {
+                  status: "flagged",
+                  createdAt: { $gt: new Date(Date.now() - 2.678e9) },
+                },
+              },
+              {
+                $unwind: "$reportedBy",
+              },
+              {
+                $set: {
+                  monthDay: {
+                    $dateToString: {
+                      format: "%d/%m",
+                      date: "$reportedBy.createdAt",
+                    },
+                  },
+                  totalCount: { $sum: 1 },
+                },
+              },
+              {
+                $group: {
+                  _id: "$monthDay",
+                  count: { $sum: 1 },
+                },
+              },
+            ],
+            KeptReportReasonCounts: [
+              {
+                $match: {
+                  status: "public",
+                  reportedBy: {
+                    $exists: true,
+                    $not: { $size: 0 },
+                  },
+                },
+              },
+              {
+                $unwind: "$reportedBy",
```

Comment:
would this (and following `unwind`) be expensive? @Naraujo13 ?

Reply:
the entire stats aggregation pipeline is very expensive, (it takes 2 seconds!) but since it is limited to a veryyyy small number of users, and will only be called once in a while, I don't think it is a concern.

Reply:
Yeah, this pipeline is expensive because of the facet (since it is basically multiple pipelines to compose this one), so the `unwind` step is not something that will impact that much performance here.
Ideally, we could switch this facet to a single pipeline, I'm not that familiar with the data being used here, but it seems that if we used a two-stage grouping we would be able to get most if not all stats without the facet (first grouping by status and reported_by, getting the number of posts in each status for each reporter, and then grouping the results by statuses only again to get the total sum). 

But this may be too much work if this is not going to scale that much.
Just creating the right indexes to support the queries, especially the initial matches, will make this drop a lot already.
I'd suggest the following indexes:

- For the stats aggregation:
      - `{ status: 1, created_at: 1 }`
      - `{ status: 1, reported_by: 1 }`
- For the new filter in the feeds posts: add the status to the index that already exists in the Post model to support this filtering (with the exact same order as the fields in the match, since the ordering is sensitive).